### PR TITLE
 fix gui2 dialog disappearing after [change_controller]

### LINKED ISSUE #2794: lua gui2 dialog sometimes no appering correctly
When playing my addon 'pick your recruits not preperation turn' on wesnoth 1.13.13, i sometimes see a screen liek the one below, where nothing is drawn until i hover the mouse over it, Sometimes the dialog also flashes shortly befroe that happens. You resizing the window make the dislog show corretly again, This usualyl happens to me on the host (which may also be becasue the host is usually the selected window at that moment when testing that ) 

grafik

### PR DESCRIPTION
**Pull request by gfgtdf**
1.14 only, no masterport, fixes #2794

### CODE REVIEW

File: src/playturn.cpp [hunk 1]
```diff
@@ -189,16 +189,21 @@ turn_info::PROCESS_DATA_RESULT turn_info::process_network_data(const config& cfg
 			resources::controller->on_not_observer();
 		}
 
+		auto disp_set_team = [&](int side_index) {
+			const bool side_chaned = display::get_singleton()->viewing_team() != side_index;
```
Comment (jyrkive) [inline src/playturn.cpp]: Typo. Should be `side_changed`.

File: src/playturn.cpp [hunk 2]
```diff
@@ -189,16 +189,21 @@ turn_info::PROCESS_DATA_RESULT turn_info::process_network_data(const config& cfg
 			resources::controller->on_not_observer();
 		}
 
+		auto disp_set_team = [&](int side_index) {
```
Comment (jyrkive) [inline src/playturn.cpp]: From what I can tell, this lambda doesn't use any local variables. Hence, don't capture them.

File: src/playturn.cpp [hunk 3]
```diff
@@ -190,7 +190,7 @@ turn_info::PROCESS_DATA_RESULT turn_info::process_network_data(const config& cfg
 		}
 
 		auto disp_set_team = [](int side_index) {
-			const bool side_changed = display::get_singleton()->viewing_team() != side_index;
+			const bool side_changed = static_cast<int>(display::get_singleton()->viewing_team()) != side_index;
```
Comment (jyrkive) [inline src/playturn.cpp]: Instead of using a cast, change the type of `side_index` to `std::size_t`.

Comment (gfgtdf) [reply src/playturn.cpp]: the cast seems fine to me.

**Comment by gfgtdf** [comment]
ok i'm merging this, i get some minor issues with chatlines appearing in the wrogn color,but i think those are caused by https://github.com/wesnoth/wesnoth/issues/2357

File: src/playturn.cpp (hunk 3 above)
Comment (jyrkive) [reply src/playturn.cpp]: Of course casting is "fine", but converting a `size_t` to `int` and casting it back later is uglier than just keeping it as `size_t`.

**Pull request merged**